### ISSUE TITLE
React-app demo does not integrate internationalization correctly

### ISSUE THREAD

**Issue by diehbria** [awslabs/iot-app-kit#929]
The react-app within IoT App Kit does not correctly utilize https://formatjs.io/docs/tooling/ts-transformer/ to provide the ids for internationalization.

When trying to utilize the react-app, due to the lack of translations IDs the user will face the following error:
![Screen Shot 2023-04-10 at 8 24 55 AM](https://user-images.githubusercontent.com/77755322/230933293-295de937-2004-4f4e-9fe9-d4e87d8756d4.png)

Expect for the scene viewer within the demo app to work without additional build system changes, as well as having this behavior publicly documented

**Comment by TheEvilDev** [comment]
Currently, some of our AppKit packages are relying on internationalization through a framework called `react-intl`. We use @formatJs's ts-transformer feature to generate Id's, which automates maintenance of these strings (new ones get added and translated automatically, old ones get removed from the code base).

To get this to work in your application, you need some minor webpack configuration following this guide:
https://formatjs.io/docs/tooling/ts-transformer/

I will take an action item to fix the Demo app to reflect this, as well as update the getting started documentation. We'll also discuss trade offs of keeping this additional requirement, or if we should use manual id's in our code base.

**Comment by TheEvilDev** [comment]
This specific issue should be fixed with: https://github.com/awslabs/iot-app-kit/pull/943. I did some deep diving, and discovered that on the 3.0 branch we didn't have this issue because we do the transformation at build time, and that simply wasn't happening anymore.

Turns out, it seems to be an issue with the version upgrades for everything, and `ttypescript` the tool we were using for this previously no longer works. I found a solution using `ts-patch`.

Note: This will break again if we try to upgrade to typescript@5, so we should avoid that upgrade until the next major version bump.
